### PR TITLE
Add ORC reader options structs to pylibcudf

### PR DESCRIPTION
**Pull request by Matt711**
## Description

Apart of #17565 
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/pylibcudf/pylibcudf/io/orc.pyx [hunk 1]
```diff
@@ -237,84 +239,190 @@ cdef class ParsedOrcStatistics:
...
+        stripes: list[list[size_type]]
+            List of lists, mapping stripes to read to input sources
+
+        Returns
+        -------
+        None
+        """
+        cdef vector[vector[size_type]] c_stripes
+        cdef vector[size_type] vec
+        for sub_list in stripes:
```
Comment (mroeschke) [inline python/pylibcudf/pylibcudf/io/orc.pyx:317]: Do we need to `c_stripes.reserve(len(stripes))` in the case of an empty `stripes`?

Comment (Matt711) [reply python/pylibcudf/pylibcudf/io/orc.pyx:317]: I think this was added originally to be a slight performance improvement (ie. you don't reallocate memory in the inner loop).

Comment (mroeschke) [reply python/pylibcudf/pylibcudf/io/orc.pyx:317]: Hmm my main concern was if `len(stripes) == 0`, we won't even hit these loops and if the uninitialized  `c_stripes` could be passed to `self.c_obj.set_stripes(c_stripes)`.

Similar to what you did in AvroReaderOptions.set_columns in https://github.com/rapidsai/cudf/pull/17599/files#diff-070361e139e5bcbddffdf4d1c8e46582455d86cf54bd364287efc555d978f900R59

Comment (Matt711) [reply python/pylibcudf/pylibcudf/io/orc.pyx:317]: Oh gotcha, I'll add a check for an empty list.

File: python/cudf/cudf/io/orc.py [hunk 2]
```diff
@@ -240,15 +240,27 @@ def read_orc(
...
+                plc.io.types.SourceInfo(filepaths_or_buffers)
+            )
+            .use_index(use_index)
+            .build()
         )
+        if num_rows >= 0:
+            options.set_num_rows(num_rows)
+        if skiprows >= 0:
+            options.set_skip_rows(skiprows)
+        if stripes is not None:
```
Comment (Matt711) [inline python/cudf/cudf/io/orc.py:254]:
```suggestion
        if stripes is not None and len(stripes) > 0:
```

**Review by mroeschke — APPROVED** [review]

**Pull request merged**